lib/opencensus/trace.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Opencensus.Trace do
+  @doc "Wraps the given block in a tracing child span with the given label/name"
+  defmacro with_child_span(label, do: block) do
+    quote do
+      :ocp.with_child_span(unquote(label), %{})
+
+      try do
+        unquote(block)
+      after
+        :ocp.finish_span()
+      end
+    end
+  end
+
+  @doc "Wraps the given block in a tracing child span with the given label/name and additional attributes"
+  defmacro with_child_span(label, attributes, do: block) do
```

Comment:
Both cases can be squashed into single definition by using:

```elixir
defmacro with_child_span(label, attributes \\ quote do: %{}, do: block) do
  # …
```

Reply:
That doesn't compile so I'm keeping it as two separate definitions for now.

Reply:
@tsloughter `quote(do: [])`, alternatively you can do:

```elixir
defmacro with_child_span(label, do: block),
  do: with_child_span(label, quote(do: []), do: block)
```